sdks/python/apache_beam/testing/load_tests/debezium_performance_test.py

```diff
@@ -0,0 +1,170 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Performance test for debezium.
+
+The purpose of this test is verify that Python's connector ReadFromDebezium
+work propertly, for this, the test create a postgresql database inside a
+kubernetes pod and stream querys inside of the database for 20 minutes.
+After that ReadFromDebezium checks if everything goes well
+
+Example test run:
+
+python -m apache_beam.testing.load_tests.debezium_performance
+
+or:
+
+./gradlew
+ -PloadTest.mainClass=apache_beam.testing.load_tests.debezium_performance \
+-Prunner=DirectRunner :sdks:python:apache_beam:testing:load_tests:run
+"""
+
+import logging
+import random
+import time
+from concurrent.futures import ThreadPoolExecutor
+
+import psycopg2
+
+from apache_beam.io.debezium import DriverClassName
+from apache_beam.io.debezium import ReadFromDebezium
+from apache_beam.testing.load_tests.load_test import LoadTest
+
+
+class DebeziumLoadTest(LoadTest):
+  def __init__(self):
+    super().__init__()
+    self.kubernetes_host = self.pipeline.get_option('kubernetes_host')
+    self.kubernetes_port = self.pipeline.get_option('kubernetes_port')
+    self.postgres_user = self.pipeline.get_option('postgres_user')
+    self.postgres_password = self.pipeline.get_option('postgres_password')
+
+    self.username = self.postgres_user
+    self.password = self.postgres_password
+
+    self.database = 'postgres'
+    self.port = self.kubernetes_port
+    self.host = self.kubernetes_host
+    self.connector_class = DriverClassName.POSTGRESQL
+    self.connection_properties = [
+        "database.dbname=postgres",
+        "database.server.name=postgres",
+        "database.include.list=postgres",
+        "include.schema.changes=false",
+        "plugin.name=pgoutput"
+    ]
+
+  def initConnection(self):
+    connection = psycopg2.connect(
+        host=self.host,
+        database=self.database,
+        user=self.username,
+        password=self.password)
+    return connection
+
+  def randomInsertTest(self):
+    connection = self.initConnection()
+    insert = 0
+    cursor = connection.cursor()
+    createTable = """
+            CREATE TABLE IF NOT EXISTS postgres(
+                id NUMERIC,
+                word VARCHAR(50),
+                number NUMERIC,
+                date DATE,
+                bool BOOLEAN
+            )
+        """
+    cursor.execute(createTable)
+    alterTableReplica = "ALTER TABLE postgres REPLICA IDENTITY FULL;"
+    cursor.execute(alterTableReplica)
+    startTime = time.time()
+    testDuration = 60 * 22
+    timeFlag = True
+
+    logging.debug('INSERTING RANDOMLY')
+    while timeFlag:
+
+      action = random.randint(1, 10)
+      if action == 1:  # Delete
```

Comment:
What is the purpose of inserting randomness here? Won't that make it harder for us to get a deterministic signal about whether performance has actually degraded?

Reply:
Hi, one of the request of the test was having a continuous random data streaming ( inserst, updates and deletes). This is the reason of having this randomness in the test. 
I agree that this approximation is not the best to get a good performance metric and could be a good change modify the behavior to get a more deterministic test.
Maybe having a fixed number of inserts, updates and deletes could be a good way.

Reply:
Yeah, I think using fixed numbers of each type is a better approach